server/src/auth/auth.service.ts

```diff
@@ -128,16 +129,17 @@ export class AuthService {
     const isDevWhileDevice =
       process.env.DEVELOPMENT === 'true' || authType !== AuthType.DEVICE;
 
-    if (!user && req.username && req.year && req.major) {
+    if (!user) {
       user = await this.userService.register(
         idToken.email,
-        req.username,
-        req.year,
-        req.major,
         req.lat,
         req.long,
         authType,
         idToken.id,
+        req.username && req.year && req.major ? true : false,
```

Comment:
does this line up with the isRegistered field? Should have a comment to make it clear